src/Components/Web/src/Forms/AntiforgeryStateProvider.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.Components.Forms;
+
+/// <summary>
+/// Provides access to the antiforgery token associated with the current session.
+/// </summary>
+public class AntiforgeryStateProvider : IDisposable
+{
+    private const string PersistenceKey = nameof(AntiforgeryRequestToken);
+    private readonly PersistingComponentStateSubscription _subscription;
+    private readonly AntiforgeryRequestToken? _currentToken;
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="AntiforgeryStateProvider"/>.
+    /// </summary>
+    /// <param name="state">The <see cref="PersistentComponentState"/> associated with the current session.</param>
+    [UnconditionalSuppressMessage(
+    "Trimming",
+    "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+    Justification = $"{nameof(AntiforgeryStateProvider)} uses the {nameof(PersistentComponentState)} APIs to deserialize the token, which are already annotated.")]
+    public AntiforgeryStateProvider(PersistentComponentState state)
+    {
+        // Automatically flow the Request token to server/wasm through
+        // persistent component state. This guarantees that the antiforgery
+        // token is available on the interactive components, even when they
+        // don't have access to the request.
```

Comment:
Do we definitely have to pass the antiforgery token to interactive components? If so, why? I'd definitely err on the side of not doing this if it's not essential.

It seems like something that would not normally be needed (interactive components normally interact with the server either via `CircuitHub` or `fetch` requests), and comes with some risk of extra disclosure, and altogether could just be considered outside the scope of this feature.

Reply:
The token would only appear in the response body if you're prerendering a form, wouldn't it?

It looks like the default here would be to serialize the antiforgery token in all cases, whether or not it's relevant (and the vast majority of the time would not be, as people have been writing WebAssembly components for years without posting to antiforgery-protected endpoints being a mainstream scenario). Apologies if I'm misreading it.

Rather than add some functionality that doesn't implement any complete end-to-end scenario out of the box, could we scope this out of the PR? We could add something like this in the future if it becomes relevant, and in the meantime people would have other ways of achieving this in the rare case they need it.

Reply:
That's because the auth in the past was done through bearer, which didn't have to account for CSRF concerns. Since we are switching to cookies for .net 8.0 we need to pass in the AF tokens, because if you make the component interactive in webassembly and call an API, you'll want to attach the token in one of the request headers.

Reply:
Thanks for the clarification.

I'm not clear on how the end-to-end programming model looks though. Can you give a code sample, showing what will be the most mainstream way to do `HttpClient`+antiforgery on WebAssembly?

Reply:
```csharp
async Task OnSubmit()
{
  var antiforgery = antiforgeryProvider.GetAntiforgeryToken();
  var request = new HttpRequestMessage(HttpMethod.Post, "action");
  request.Headers.Add("RequestVerificationToken", antiforgery.RequestToken);
  var response = await client.SendAsync(request);
  ...
}
```

Reply:
https://learn.microsoft.com/en-us/aspnet/core/security/anti-request-forgery?view=aspnetcore-7.0#javascript-ajax-and-spas

Reply:
Is that what you'd expect to be the typical guidance, as opposed to something like an `HttpMessageHandler` that attaches the token automatically?

Reply:
I would expect the guidance to showcase both patterns, with a preference for the message handler.

Reply:
> ```cs
> async Task OnSubmit()
> {
>   var antiforgery = antiforgeryProvider.GetAntiforgeryToken();
>   var request = new HttpRequestMessage(HttpMethod.Post, "action");
>   request.Headers.Add("RequestVerificationToken", antiforgery.RequestToken);
>   var response = await client.SendAsync(request);
>   ...
> }
> ```

This seems *excessively* cumbersome for an API call. Will there be extensions or some feature to wrap all this into a single call?